### PR TITLE
Plugins: Add better plugin notices to install and update errors

### LINKED ISSUE #409: Jetpack Plugins: Give a more specific error message on plugin activation/deactivation/update failures
If you try to activate a plugin that causes a fatal error, we don't currently pass that detail back from the remote site (or display it in Calypso). We indicate that the update failed, but not really _why_. We should ideally pass back a more descriptive reason (from the remote site), and then provide an error along the lines of what WP core does ("Plugin could not be activated because it triggered a fatal error.")

Here's a detailed example: Test on a site that requires FTP permission in order to install or update plugins. This is accomplished by setting the file permissions of wp-content to 555 (the directory itself, not necessarily the files within).

To make sure that FTP permissions are needed to install a plugin, you can try installing a plugin and get this dialog:

install-plugin-get-asked-for-ftp-deets

You can try to update a plugin and get this dialog:

update-plugin-need-ftp

When trying to update a plugin on Calypso you see an error message instead of nothing:

Step 1: Click the update link:

d96d0142-7e51-11e5-80e8-acf0f99649e8

Step 2: Calypso shows an alert that it's trying to update the plugin:

fd858f7c-7e51-11e5-81a3-bbd9cf1699bf

Step 3: see error message:

0e4bad46-7e52-11e5-9a3f-01f90c1a07d4

The error message isn't very informative.

### PR DESCRIPTION
**Pull request by enejb**
Fixes #409. 

Add additional explanation As to why a plugin might have failed or updated. The explanation text currently is exactly the same that we would get from WordPress if the same error occurred. I wasn't able to test each of the cases. 

cc: @johnHackworth, @roccotripaldi and @lezama

### CODE REVIEW

**Comment by johnHackworth** [comment]
Works great for the cases I could test! :ship: 

image

File: client/lib/plugins/notices.jsx [hunk 1]
```diff
@@ -550,7 +550,83 @@ module.exports = {
...
+				return i18n.translate( 'Could not create directory.' );
+
+			case 'incompatible_archive':
+				return i18n.translate( 'The package could not be installed.' );
+
+			case 'files_not_writable':
+				return i18n.translate( 'The update cannot be installed because we will be unable to copy some files. This is usually due to inconsistent file permissions.' );
+
+			default:
+				return null;
```
Comment (roccotripaldi) [inline client/lib/plugins/notices.jsx]: Nice, exhaustive list!

File: client/lib/plugins/notices.jsx [hunk 2]
```diff
@@ -550,7 +550,83 @@ module.exports = {
...
+				return i18n.translate( 'Unable to locate WordPress Plugin directory.' );
+
+			case 'fs_no_themes_dir':
+				return i18n.translate( 'Unable to locate WordPress Theme directory.' );
+
+			case 'fs_no_folder':
+				return i18n.translate( 'Unable to locate needed folder.' );
+
+			case 'download_failed':
+				return i18n.translate( 'Download failed.' );
```
Comment (roccotripaldi) [inline client/lib/plugins/notices.jsx]: I wonder if this should be `no_package` instead? as we updated jetpack to use `no_package` for backwards compatibility.
https://github.com/Automattic/jetpack/blob/master/json-endpoints/jetpack/class.jetpack-json-api-plugins-install-endpoint.php#L42

**Comment by roccotripaldi** [comment]
Looks good, thanks for picking this up Enej, and sorry I could not get to it.
I left one comment. I was only able to test one case: 'Plugin already installed'
In trying to emulate other cases like `fs_no_plugins_dir` or `files_not_writable`, would cause wordpress to error out early and leave me with the generic error message before a more verbose message could be output. I think we may be stuck with that.

File: client/lib/plugins/notices.jsx (hunk 2 above)
Comment (enejb) [inline client/lib/plugins/notices.jsx]: We have `no_package` one above ^.

Comment (roccotripaldi) [inline client/lib/plugins/notices.jsx]: o_o

**Pull request merged**